front/app/hooks/useAdminPublications.ts

```diff
@@ -78,12 +81,27 @@ export default function useAdminPublications({
   const [loadingMore, setLoadingMore] = useState(false);
   const [pageNumber, setPageNumber] = useState(1);
   const [search, setSearch] = useState<string | null>(null);
-  const [topics, setTopics] = useState<string[] | undefined>(topicFilter);
-  const [areas, setAreas] = useState<string[] | undefined>(areaFilter);
+  const [topics, setTopics] = useState<string[] | null>(null);
+  const [areas, setAreas] = useState<string[] | null>(null);
   const [publicationStatuses, setPublicationStatuses] = useState<
     PublicationStatus[]
   >(publicationStatusFilter);
 
+  // topicFilter and areaFilter are usually based of other
+  // requests, and will initially be null/undefined.
+  // Without the useEffect, they don't get updated
+  // (or useDeepCompareEffect because we are comparing arrays here).
+  useDeepCompareEffect(() => {
+    if (topicFilter !== undefined) {
+      setTopics(topicFilter);
+    }
+  }, [topicFilter]);
+  useDeepCompareEffect(() => {
```

Comment:
Wanted to check on the mechanism here, I don't think this library is necessary. From their docs: `NOTE: Only use this if your values are objects or arrays that contain objects. Otherwise, you should just use React.useEffect`. Since topic/area filters are an array of string ID values, you could just use `JSON.stringify` and put the stringified value into the dependency array. We do this in a few places already (`useTopics.ts` as one example)

Reply:
Yeah, just checking because I'll need the same changes to make sure the admin-facing list works the same as the citizen-facing one. I messed a bit with this hook as well, theoretically json.stringify should work but the rest of the hook is pretty complicated so it's possible something else interferes

Reply:
One issue is that parsing it returns `any` as a type.

E.g. `useEvents.ts`:

```
  const stringifiedProjectIds = JSON.stringify(parameters.projectIds);
  useEffect(() => {
    const projectIds = JSON.parse(stringifiedProjectIds);
    if (projectIds) {
      setProjectIds(JSON.parse(projectIds));
    }
  }, [stringifiedProjectIds]);
```

If `projectIds` is `undefined` here the `const projectIds = JSON.parse(stringifiedProjectIds);` line will cause an uncaught SyntaxError and the page crashes.

Reply:
I don't think you have to parse them to/from JSON to actually within the component, the stringified prop is just to create a value for the dependency array that can be easily compared by `useEffect`

Reply:
It works in other places for different reasons:
- The parameter to stringify is never undefined
- The stringified parameter is not used inside the useEffect. The linter is then silenced with `// eslint-disable-next-line react-hooks/exhaustive-deps`, of which I'm not a fan either

Reply:
> I don't think you have to parse them to/from JSON to actually within the component, the stringified prop is just to create a value for the dependency array that can be easily compared by useEffect

Then the linter will complain, so you have to silence it and it'll lose its value.

Reply:
The linter gives this advice:

> If 'setProjectIds' needs the current value of 'parameters.projectIds', you can also switch to useReducer instead of useState and read 'parameters.projectIds' in the reducer

Reply:
I'm not sure about that one, just referring to the topic filter. If the goal is to have the `useEffect` run whenever the filter changes, stringifying it and running the effect whenever the output string changes should be fine. You can silence the linter rule (as we already do in quite a few places) because you specifically only want the dependency array to check and rerun the effect for any changes in the stringified array of IDs, and nothing else

Reply:
Agreed that we should go with that solution now (silence linter) or we could be digging for quite a while.

I thought Iva said we want to get rid of that rule. So definitely CDM material and platform work if we're going to tackle this. Will add it to my list to log when the epic is finished.

Reply:
I added it to the next FE-CDM 👍

Reply:
I'll make sure in useEvents to turn undefined projectIds into null so it doesn't crash. Hacky stuff. 🏴‍☠️